tests/filter.rs

```diff
@@ -85,7 +85,7 @@ fn filter_parent_with_matched_child() {
     setup();
 
     select_and_then_compare(
-        "$.a[?(@.b.c == 1)]",
+        "$[?(@.b.c == 1)]",
```

Comment:
Consider also keeping the previous path (`"$.a[?(@.b.c == 1)]"`) and test it returns nothing  (`json!([])`)

Reply:
Fixed